mmv1/products/dialogflowcx/api.yaml

```diff
@@ -111,3 +111,114 @@ objects:
         name: 'enableSpellCorrection'
         description: |
          Indicates if automatic spell correction is enabled in detect intent requests.
+  - !ruby/object:Api::Resource
+    name: 'Version'
+    base_url: "{{parent}}/versions/"
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      You can create multiple versions of your agent flows and deploy them to separate serving environments.
+      When you edit a flow, you are editing a draft flow. At any point, you can save a draft flow as a flow version. A flow version is an immutable snapshot of your flow data and associated agent data like intents, entities, webhooks, pages, route groups, etc.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/dialogflow/cx/docs'
+      api: 'https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents.flows.versions'
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        full_url: 'https://dialogflow.googleapis.com/v3/{{op_id}}'
```

Comment:
hi @rileykarson, yes I noticed that but I don't found a way to set 2 differents base_url to the same product. Because for Agents we need base_url with location and for all the others resources (version, environments, entities,etc) we require base_url without location.
I think so an option could be If I set base_url in product as https://dialogflow.googleapis.com/ (without location) and then I include custom code in Agent create/delete/update to include location in url. Because finally most of resources don't need regional endpoint

Or maybe you have another idea?

Reply:
Where I'm confused is that the API document at https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents/create suggests agent has a global endpoint as well, given `POST https://dialogflow.googleapis.com/v3/{parent=projects/*/locations/*}/agents` doesn't point to `{{location}}-dialogflow`. How did you find out a regional endpoint was required for that resource?

Reply:
got it, you are right, I don't know if was an update in API because previously I remember I saw in documentation and was a regional endpoint. But in this way is more easy to manage a global endpoint.
I changed API to use https://dialogflow.googleapis.com/v3/ instead and it seems like agents is working fine

Thanks